src/EPR.Calculator.Service.Function/Services/CalculatorRunService.cs

```diff
@@ -1,56 +1,161 @@
+                    Configuration.OrgDataPipelineName);
 
-            var pomPipelineConfiguration = GetAzureSynapseConfiguration(
-                calculatorRunParameter,
-                Configuration.PomDataPipelineName);
-            var isPomSuccessful = await azureSynapseRunner.Process(pomPipelineConfiguration);
+                var isOrgSuccessful = await this.azureSynapseRunner.Process(orgPipelineConfiguration);
 
-            this.logger.LogInformation("Pom status", isPomSuccessful);
+                this.logger.LogInformation("Org status: {Status}", Convert.ToString(isOrgSuccessful));
 
-            var orgPipelineConfiguration = GetAzureSynapseConfiguration(
-                calculatorRunParameter,
-                Configuration.OrgDataPipelineName);
+                if (isOrgSuccessful)
+                {
+                    var pomPipelineConfiguration = GetAzureSynapseConfiguration(
+                        calculatorRunParameter,
+                        Configuration.PomDataPipelineName);
+                    isPomSuccessful = await this.azureSynapseRunner.Process(pomPipelineConfiguration);
 
-            var isOrgSuccessful = await azureSynapseRunner.Process(orgPipelineConfiguration);
+                    this.logger.LogInformation("Pom status: {Status}", Convert.ToString(isPomSuccessful));
+                }
+            }
+            else
+            {
+                isPomSuccessful = true;
+            }
 
-            this.logger.LogInformation("Org status", isOrgSuccessful);
-        }
+            this.logger.LogInformation("Pom status: {Status}", Convert.ToString(isPomSuccessful));
 
-        private static AzureSynapseRunnerParameters GetAzureSynapseConfiguration(
-            CalculatorRunParameter args,
-            string pipelineName)
-            => new AzureSynapseRunnerParameters()
+            using var client = this.pipelineClientFactory.GetHttpClient(Configuration.StatusEndpoint);
+
+            if (isPomSuccessful)
             {
-                PipelineUrl = new Uri(Configuration.PipelineUrl),
-                CheckInterval = int.Parse(Configuration.CheckInterval),
-                MaxChecks = int.Parse(Configuration.MaxCheckCount),
-                PipelineName = pipelineName,
-                CalculatorRunId = args.Id,
-                FinancialYear = Util.GetFinancialYearAsYYYY(args.FinancialYear),
-                StatusUpdateEndpoint = Configuration.StatusEndpoint,
-            };
+                var prepareCalcResultResponse = await client.PostAsync(
+                    Configuration.PrepareCalcResultEndPoint,
+                    GetPrepareCalcResultMessage(calculatorRunParameter.Id));
+                isSuccess = prepareCalcResultResponse.IsSuccessStatusCode;
+            }
+
+            var statusUpdateResponse = await client.PostAsync(
+                Configuration.StatusEndpoint,
+                GetStatusUpdateMessage(calculatorRunParameter.Id, isSuccess));
+
+#if DEBUG
```

Comment:
Please remove Debug.Writeline statements now as we have added Logs

Reply:
Resolved in PR#16